services/traction/api/db/models/base.py

```diff
@@ -36,3 +81,28 @@ class BaseTable(BaseModel):
             TIMESTAMP, nullable=False, server_default=func.now(), onupdate=func.now()
         )
     )
+
+
+class TimestampModel(BaseModel):
+    created_at: datetime = Field(
+        sa_column=Column(TIMESTAMP, nullable=False, server_default=func.now())
+    )
+    updated_at: datetime = Field(
+        sa_column=Column(
+            TIMESTAMP, nullable=False, server_default=func.now(), onupdate=func.now()
+        )
+    )
+
+
+class StatefulModel(BaseModel):
+    status: str = Field(nullable=False)
+    state: str = Field(nullable=False)
+    error_status_detail: str = Field(nullable=True)
+
+
+class TimelineModel(BaseModel):
```

Comment:
are these two related? can you have a TimelineModel without it also being also StatefulModel?

Reply:
They are different tables, Timeline is just tracking history of status it's strictly audit, not a business object table. you could add error_status_detail if wanted, but a lot of the status updates are just coming from acapy messages.

Reply:
ah I see now, makes sense.

Reply:
we could still throw the error_status_detail in the timeline... may be good to see there what the error was, even if it gets corrected.